tests/test_fix_known.py

```diff
@@ -0,0 +1,272 @@
+        """Check that the constructed test cube coords are not contiguous"""
+        for coord_name in ['latitude', 'longitude']:
+            coord = self.cube.coord(coord_name)
+            self.assertFalse(coord.is_contiguous())
+
+    def test_bounds_contiguous(self):
+        fix = ch.fix_known.FixCmip6CasFgoals(self.cube)
+        fix.fix_cube()
+        for coord_name in ['latitude', 'longitude']:
+            coord = self.cube.coord(coord_name)
+            self.assertTrue(coord.is_contiguous())
+
+
+class TestFixCmip6FioqlnmFioesm20Historical(TestCase):
+    """Test cube_helper.fix_known.FixCmip6FioqlnmFioesm20Historical"""
+    def setUp(self):
+        self.cube = _make_fioesm20_historical_cube()
+
+        patch = mock.patch('cube_helper.fix_known.log_module')
+        self.mock_logger = patch.start()
+        self.addCleanup(patch.stop)
+
+    def test_is_fix_needed(self):
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        self.assertTrue(fix.is_fix_needed())
+
+    def test_is_fix_needed_fails_mip_era(self):
+        self.cube.attributes['mip_era'] = 'ABCD'
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        self.assertFalse(fix.is_fix_needed())
+
+    def test_is_fix_needed_fails_institution_id(self):
+        del self.cube.attributes['institution_id']
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        self.assertFalse(fix.is_fix_needed())
+
+    def test_is_fix_needed_fails_source_id(self):
+        self.cube.attributes['source_id'] = '1234'
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        self.assertFalse(fix.is_fix_needed())
+
+    def test_is_fix_needed_fails_experiment_id(self):
+        self.cube.attributes['experiment_id'] = '1234'
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        self.assertFalse(fix.is_fix_needed())
+
+    def test_log_message(self):
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        fix.fix_cube()
+        fix.logger.info.assert_called_with('Applying FixCmip6FioqlnmFioesm20'
+                                           'Historical')
+
+    def test_constructed_cube_not_monotonic(self):
+        """Check that the constructed test cube latitude is not monotonic"""
+        self.assertFalse(self.cube.coord('latitude').is_monotonic())
+
+    def test_constructed_cube_dim_coords(self):
+        """Check that the constructed test cube latitude is not dim coord"""
+        dim_coord_names = [dim_coord.standard_name
+                           for dim_coord in self.cube.dim_coords]
+        expected_names = ['time', 'longitude']
+        self.assertEqual(dim_coord_names, expected_names)
+
+    def test_bounds_monotonic(self):
+        fix = ch.fix_known.FixCmip6FioqlnmFioesm20Historical(self.cube)
+        fix.fix_cube()
+        coord = self.cube.coord('latitude')
+        self.assertTrue(coord.is_monotonic())
```

Comment:
Would it be possible to also test that the longitude bounds are monotonic? The documentation for the `FixCmip6FioqlnmFioesm20Historical` class (and the code in the `fix_cube` method) indicates that the longitude bounds are also fixed.

Reply:
I'm not sure that this is necessary as the longitude bounds are already monotonic (longitude was able to be a dimension coordinate, whereas latitude was only an aux coord on load). I've added a better explanation in the docstrings in 3c7d1f0. While I was doing this I also noticed that the specification that I was given for this class was wrong and it affected a random selection of all FIO-ESM-2-0 variables rather than just historic, and so the identification (and tests and documentation) has been updated to reflect this.

Reply:
As far as I can tell the only thing that hasn't been tested is that the longitude bounds are updated. Is it possible to include a test that shows that the longitude bounds are also updated?

Reply:
Added test with 3ea6198.